src/api/ledger/transactions.js

```diff
@@ -28,8 +28,15 @@ function transactionFilter(address, filters, tx) {
   if (filters.initiated === false && tx.address === address) {
     return false;
   }
-  if (filters.counterparty && tx.address !== filters.counterparty
-      && tx.specification.destination.address !== filters.counterparty) {
+  if (filters.counterparty) {
+    if (tx.address === filters.counterparty || (
+      tx.specification && (
+        (tx.specification.destination &&
+          tx.specification.destination.address === filters.counterparty) ||
+        (tx.specification.counterparty === filters.counterparty)
+      ))) {
+      return true;
```

Comment:
It's a bit risky to put a `return true` here because someone might add another condition after this one. Could you create a new function to check for counterparty match so that it can keep the same pattern of only returning false?